scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/TaskLauncher.java

```diff
@@ -365,12 +380,19 @@ private void sendResultToScheduler(TaskTerminateNotification terminateNotificati
         int pingAttempts = initializer.getPingAttempts();
         int pingPeriodMs = initializer.getPingPeriod() * 1000;
 
+        TaskTerminateNotification currentTerminateNotification = terminateNotification;
         for (int i = 0; i < pingAttempts; i++) {
             try {
-                terminateNotification.terminate(taskId, taskResult);
+                currentTerminateNotification.terminate(taskId, taskResult);
                 logger.debug("Successfully notified task termination " + taskId);
                 return;
             } catch (Throwable t) {
+                logger.warn("Cannot notify task termination, trying to rebind to to the task termination handler", t);
```

Comment:
This method is too complicated in terms of try/catch/if/for/continue. It should be splitted into several methods

Reply:
Addressed.